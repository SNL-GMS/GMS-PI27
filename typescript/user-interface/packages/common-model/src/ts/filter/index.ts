export * from './filter-util';
export * from './types';
