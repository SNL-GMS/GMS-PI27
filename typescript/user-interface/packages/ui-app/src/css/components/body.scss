body,
body.bp5-dark {
  background-color: var(--gms-background, $gms-background);
}

.bp5-navbar {
  padding: 0;
}

.bp5-control input:focus ~ .bp5-control-indicator {
  outline: none;
  outline-offset: 0px;
}
.bp5-control {
  margin-bottom: 0px;
}
.bp5-control:not(.checkbox-horizontal) + .bp5-control:not(.checkbox-horizontal) {
  margin-top: 8px;
}
.bp5-control.checkbox-horizontal + .bp5-control.checkbox-horizontal {
  margin-left: 8px;
}
.bp5-dark .bp5-control.bp5-switch .bp5-control-indicator {
  background: var(--gms-popover-background, $gms-popover-background);
  margin-left: 4px;
}
.bp5-dark .bp5-control.bp5-switch .bp5-control-indicator::before {
  background: var(--gms-main, $gms-main);
}
.bp5-dark .bp5-control.bp5-switch:hover .bp5-control-indicator {
  background: var(--gms-soft, $gms-soft);
}

.bp5-dark .bp5-button:not([class*='bp5-intent-']) {
  background: var(--gms-button, $gms-button);
}
.bp5-dark .bp5-button:not([class*='bp5-intent-']):hover {
  background: var(--gms-button-hover, $gms-button-hover);
}
.bp5-dark .bp5-button.bp5-minimal {
  background: none;
  background-color: none;
}

.bp5-dialog:not(:has(.bp5-dialog-footer)) {
  padding-bottom: 0.5rem;
}

.bp5-dark .bp5-dialog {
  background-color: var(--gms-dialog-background, $gms-dialog-background);
}

.bp5-dialog-header {
  box-shadow: none;
}

.bp5-dark .bp5-dialog-header {
  box-shadow: none;
}
*::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}

*::-webkit-scrollbar-track {
  background-color: var(--gms-scrollbar-background, $gms-scrollbar-background);
  width: 10px;
}

*::-webkit-scrollbar-thumb {
  background: var(--gms-scrollbar, $gms-scrollbar);
  border-radius: 4px;
}

:focus {
  outline: none;
  outline-offset: 0px;
}

.max {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
}

.duo-grid-left {
  position: absolute;
  top: 22%;
  right: 0%;
  left: 1.5%;
  width: 48%;
  height: 18%;
}

.duo-grid-right {
  position: absolute;
  top: 22%;
  right: 0%;
  left: 52%;
  width: 48%;
  height: 18%;
}

.bp5-menu {
  min-width: auto !important;
}

.gms-disconnected {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 100;
  background-color: rgba(var(--gms-background, $gms-background), 0.85);
  display: flex;
  justify-content: center;
  align-items: center;
}
