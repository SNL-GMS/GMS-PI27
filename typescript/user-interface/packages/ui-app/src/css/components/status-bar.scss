.status-bar {
  container-type: inline-size;
  container-name: statusbar;
  background-color: var(--gms-table-header-background, $gms-table-header-background);
  border-top: 1px solid var(--gms-recessed, $gms-recessed);
  bottom: 0;
  display: flex;
  align-items: center;
  font-size: 1rem;
  font-weight: 300;
  gap: 0.5em;
  height: calc(1.75rem - 1px); // handle border size
  letter-spacing: 1px;
  position: relative;
  width: 100%;
  z-index: 1;
  overflow-y: hidden;
  @include addBoxShadow('lg');
}

.loading-counter {
  align-items: center;
  border-radius: 0.25rem;
  display: flex;
  font-family: $gms-mono;
  justify-content: center;
}

.loading-widget {
  overflow: hidden;
  white-space: nowrap;
  align-items: center;
  display: flex;
  gap: 0.5rem;
  height: 100%;
  padding: 0 0.5rem;
  position: relative;

  &:hover {
    cursor: pointer;
    @include castLightOn(0.1);
  }

  &:hover .bp5-icon-issue {
    color: red;
  }
}

.selection-information-container {
  margin-left: auto;
}

.selection-information-widget__text {
  display: block;
}

.selection-information-widget__text--medium {
  display: none;
}

@container statusbar (max-width: 1300px) {
  .selection-information-widget__text {
    display: none;
  }
  .selection-information-widget__text--medium {
    display: block;
  }
}

@container statusbar (max-width: 850px) {
  .selection-information-widget__text {
    display: none;
  }
  .selection-information-widget__text--medium {
    display: none;
  }
}

.selection-information-widget {
  align-items: center;
  display: flex;
  gap: 0.5rem;
  height: 100%;
  padding: 0 0.5rem;
  position: relative;
  font-family: $gms-mono;
  &:hover {
    cursor: pointer;
    @include castLightOn(0.1);
  }

  &--disabled {
    color: var(--gms-soft, $gms-soft);
    &:hover {
      cursor: not-allowed;
    }
  }
}

.loading-info {
  font-size: 1rem;
  font-weight: 300;
  letter-spacing: 1px;

  &__label {
    font-weight: 500;
  }

  &__loading-info {
    display: flex;
    font-size: 0.875rem;
    gap: 0.5rem;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}

.loading-list {
  --gms-scrollbar-background: lightgray;
  --gms-scrollbar: gray;
  display: flex;
  flex-direction: column-reverse;
  gap: 0.25rem;
  list-style-type: none;
  margin-block-end: 0;
  margin-block-start: 0;
  margin-inline-end: 0;
  margin-inline-start: 0;
  max-height: 30em;
  min-width: 40ch;
  overflow-y: auto;
  padding-inline-start: 0;
  padding: 0.5rem 0.5rem 0 0;

  &__sticky-label {
    border-bottom: 1px solid var(--gms-main-inverted, $gms-main-inverted);
    font-size: 1.25em;
    font-weight: 600;
    padding: 0.25rem 0.5rem 0.25rem 0;
    position: sticky;
    top: 0;
  }
}

.loading-list-entry {
  position: relative;

  &__description {
    padding-left: 1.25rem;
    position: relative;

    &--set-apart {
      padding-bottom: 0.25rem;
      padding-top: 0.25rem;
    }
  }

  &__label {
    font-weight: 600;
  }

  &.is-error {
    color: darkred;
  }

  &__icon,
  &__spinner {
    left: 0;
    position: absolute;
    top: 50%;
    transform: translate3D(0, -50%, 0);
  }

  &__more-info {
    display: block;
    padding-left: 1.25rem;
  }
}
