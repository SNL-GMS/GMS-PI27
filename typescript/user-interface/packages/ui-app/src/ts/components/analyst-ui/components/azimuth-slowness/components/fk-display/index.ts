export { FkDisplay } from './fk-display';
