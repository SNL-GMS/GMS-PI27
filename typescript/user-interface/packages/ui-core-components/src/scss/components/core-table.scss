/// Blueprint Table
.core-table__container {
  .bp5-table-header .bp5-table-column-name {
    height: 100%;
  }
  .bp5-table-header .bp5-table-header-content {
    position: absolute;
    margin-left: 24px;
    top: 0;
    display: inline-flex;
  }
  .core-table__column-header {
    display: flex;
    align-items: center;
    width: calc(100% - 25px);
    & .bp5-icon.bp5-intent-primary {
      color: var(--core-main, var(--gms-main, $core-main));
    }
    .force-right {
      margin-left: auto;
      margin-right: 0.5em;
    }
  }

  /// Tooltips
  .bp5-table-cell .bp5-popover-target {
    width: 100%;
  }
}
.core-table__column-header--default {
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
}
