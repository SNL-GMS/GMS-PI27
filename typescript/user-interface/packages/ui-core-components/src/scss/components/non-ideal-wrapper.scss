.non-ideal-loading-wrapper-hidden {
  height: 0%;
  width: 0%;
  visibility: hidden;
}

.non-ideal-loading-wrapper {
  height: 100%;
  width: 100%;
}
