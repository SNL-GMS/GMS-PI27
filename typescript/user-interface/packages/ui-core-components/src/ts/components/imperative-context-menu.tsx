import type { ContextMenuPopoverProps } from '@blueprintjs/core';
import { Classes, ContextMenuPopover, hideContextMenu } from '@blueprintjs/core';
import { uuid4 } from '@gms/common-util';
import { UILogger } from '@gms/ui-util';
import defer from 'lodash/defer';
import React from 'react';
import type { Root } from 'react-dom/client';
import { createRoot } from 'react-dom/client';

const logger = UILogger.create(
  'GMS_IMPERATIVE_CONTEXT_MENU',
  process.env.GMS_IMPERATIVE_CONTEXT_MENU
);

// the imperative context menu container; a singleton instance
let imperativeContainer: ImperativeContainer | undefined;

/** manages the deferred hide/close operations */
let deferHide: { func: () => void; timerId: number | undefined } | undefined;

/**
 * The properties of an imperative context menu.
 */
interface ImperativeContainer {
  /**
   * The unique id of the context menu used to determine changes; e.g. opening one context menu from another
   */
  readonly id: string;

  /**
   * The container for the context menu
   */
  readonly rootContainer: Root;

  /**
   * The internal {@link HTMLElement} container for the context menu
   */
  readonly container: HTMLElement;

  /**
   * The active element with focus prior to opening up the context menu
   */
  readonly activeElement: HTMLElement;

  /**
   * A reference to the onClose method that was passed to the imperative context
   */
  readonly onClose?: () => void;
}

/**
 * Imperative Context Menu Props
 */
export type ImperativeContextMenuProps = Omit<ContextMenuPopoverProps, 'isOpen'> & {
  /**
   * (optional) if set the element will receive focus on close
   */
  readonly activeElementOnClose?: HTMLElement;
};

/**
 * Hide Imperative Context Options
 */
export interface HideImperativeContextMenuOptions {
  /**
   * (optional) if set to true; then the context menu onClose callback will be invoked
   * ! default is set to true
   */
  readonly shouldInvokeOnClose?: boolean;

  /**
   * (optional) if set to true; then the context menu will return focus on close
   *
   * @default true
   */
  readonly shouldReturnFocus?: boolean;

  /**
   * (optional) an optional callback to be called after the imperative context menu has been closed
   * ? This callback will execute after the context menu onClose (if enabled) and the refocus of the active element (if enabled)
   *
   * @default undefined
   */
  readonly callback?: () => void;
}

/** helper to gather the props needed for hiding the imperative context menu */
function getHideImperativeContextMenuProps(options: HideImperativeContextMenuOptions = {}) {
  return {
    shouldInvokeOnClose: options.shouldInvokeOnClose ?? true,
    shouldReturnFocus: options.shouldReturnFocus ?? true,
    callback: options.callback || undefined,
    activeElement: imperativeContainer?.activeElement,
    onClose: imperativeContainer?.onClose
  };
}

/** unmount the imperative context menu  */
function unmountImperativeContextMenu() {
  if (imperativeContainer !== undefined) {
    imperativeContainer.container.remove();
    imperativeContainer.rootContainer.unmount();
    imperativeContainer = undefined;
  }
}

/** remove the context menu content from the dom */
function removeContextMenuFromDom() {
  const elements = [
    ...document.getElementsByClassName(`${Classes.POPOVER_CONTENT}`),
    ...document.getElementsByClassName(`${Classes.POPOVER_BACKDROP}`),
    ...document.getElementsByClassName(`${Classes.TOOLTIP}`),
    ...document.getElementsByClassName(`${Classes.POPOVER_TRANSITION_CONTAINER}`)
  ];

  elements.forEach((e, idx) => {
    elements[idx].innerHTML = '';
  });
}

/**
 * Hide imperative context menu that was created using {@link showImperativeContextMenu}.
 *
 * @see https://blueprintjs.com/docs/#core/components/context-menu-popover.imperative-api
 */
export function hideImperativeContextMenu(options: HideImperativeContextMenuOptions = {}) {
  // if multiple hide operations are called immediately; bypass the defer and execute the previous call first
  if (deferHide !== undefined) {
    logger.debug('forcing execution of previous deferred hide');
    clearTimeout(deferHide.timerId);
    deferHide.func();
  }

  logger.debug('hide invoked');

  const { shouldInvokeOnClose, shouldReturnFocus, callback, activeElement, onClose } =
    getHideImperativeContextMenuProps(options);

  deferHide = {
    timerId: undefined,
    func: () => {
      if (shouldInvokeOnClose && onClose) {
        onClose();
      }
      if (callback) {
        callback();
      }

      if (shouldReturnFocus && activeElement) {
        activeElement.focus();
      }

      deferHide = undefined;
    }
  };

  // unmount the imperative context menu
  unmountImperativeContextMenu();

  hideContextMenu();

  // remove the context menu content from the dom
  removeContextMenuFromDom();

  deferHide.timerId = defer(deferHide.func);
}

/**
 * A simple wrapper around {@link ContextMenu2Popover} which is open by default and controlled.
 * The context menu closes when a user clicks outside the popover.
 */
function UncontrolledContextMenuPopover(props: ImperativeContextMenuProps) {
  const [isOpen, setIsOpen] = React.useState(true);

  const idRef = React.useRef(imperativeContainer?.id);

  const handleClose = React.useCallback(() => {
    logger.debug('on close invoked');
    setIsOpen(false);

    if (imperativeContainer?.id !== idRef.current) {
      logger.debug('no need to handle on close');
    } else {
      logger.debug('handle on close');
      hideImperativeContextMenu();
    }
  }, []);

  //! useEffect updates local state
  React.useEffect(() => {
    return () => {
      setIsOpen(false);
    };
  }, []);

  // eslint-disable-next-line react/jsx-props-no-spreading
  return <ContextMenuPopover key={uuid4()} isOpen={isOpen} {...props} onClose={handleClose} />;
}

/**
 * Show Imperative Context Options
 */
export interface ShowImperativeContextMenuOptions {
  /**
   * (optional) if set to true; then the context menu onClose callback will be invoked
   * ! default is set to false
   */
  readonly shouldInvokeOnClose?: boolean;

  /**
   * (optional) if set to true; then the context menu will return focus on close
   * ! default is set to true
   */
  readonly shouldReturnFocus?: boolean;
}

/**
 * Show imperative context menu.
 *
 * Shows a context menu at a particular offset from the top-left corner of the document.
 * The menu will appear below-right of this point and will flip to below-left if there is not enough
 * room onscreen. Additional props like `onClose`, `isDarkTheme`, etc. can be forwarded to the {@link ContextMenu2Popover}.
 *
 * Context menus created with this API will automatically close when a user clicks outside the popover.
 * You may force them to close by using {@link hideImperativeContextMenu}.
 *
 * @see https://blueprintjs.com/docs/#core/components/context-menu-popover.imperative-api
 */
export function showImperativeContextMenu(
  props: ImperativeContextMenuProps,
  options: Omit<HideImperativeContextMenuOptions, 'callback'> = {}
) {
  logger.debug('show invoked');

  const shouldInvokeOnClose = options.shouldInvokeOnClose ?? false;
  const shouldReturnFocus = options.shouldReturnFocus ?? true;
  const previousActiveElement: HTMLElement | undefined = imperativeContainer?.activeElement;

  hideImperativeContextMenu({
    shouldInvokeOnClose,
    shouldReturnFocus,
    callback: () => {
      // show the new context menu after ensuring that the previous one has been closed and cleaned up
      const { onClose } = props;

      const id: string = uuid4();

      const container: HTMLElement = document.createElement(`imperative-context-menu`);
      container.id = id;
      container.classList.add(Classes.CONTEXT_MENU);
      document.body.appendChild(container);

      const rootContainer: Root = createRoot(container, {
        identifierPrefix: 'imperative-context-menu'
      });

      const activeElement: HTMLElement =
        props.activeElementOnClose ||
        previousActiveElement ||
        (document.activeElement as HTMLElement);

      imperativeContainer = {
        id,
        rootContainer,
        container,
        activeElement,
        onClose
      };

      imperativeContainer.rootContainer.render(
        // eslint-disable-next-line react/jsx-props-no-spreading
        <UncontrolledContextMenuPopover key={id} {...props} />
      );

      /**
       * The {@link ContextMenu2Popover} has a boarder that if clicked can allow the default system context menu to popup.
       *
       * !Once the {@link ContextMenu2Popover} is rendered (defer) add an event listener to capture and prevent the default system context menu from showing.
       */
      defer(() => {
        const elements = [
          ...document.getElementsByClassName(`${Classes.CONTEXT_MENU_VIRTUAL_TARGET}`),
          ...document.getElementsByClassName(`${Classes.CONTEXT_MENU_BACKDROP}`),
          ...document.getElementsByClassName(`${Classes.CONTEXT_MENU}`)
        ];
        elements.forEach(element => {
          element.addEventListener('contextmenu', (event: Event) => {
            event.preventDefault();
          });
        });
      });
    }
  });
}
