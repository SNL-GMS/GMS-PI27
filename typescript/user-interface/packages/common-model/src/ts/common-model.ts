import * as ArrayUtil from './array-util';
import * as BeamformingTemplateTypes from './beamforming-templates/types';
import * as ChannelSegmentTypes from './channel-segment';
import * as ColorTypes from './color/types';
import * as CommonTypes from './common';
import * as Displays from './displays/types';
import * as Endpoints from './endpoints/types';
import * as EventTypes from './event';
// required for legacy components
import * as LegacyEventTypes from './event/legacy';
import * as FacetedTypes from './faceted';
import * as FilterUtil from './filter/filter-util';
import * as FilterTypes from './filter/types';
import * as FkTypes from './fk';
import * as ProcessingMaskDefinitionTypes from './processing-mask-definitions/types';
import * as QcSegmentTypes from './qc-segment';
import * as RotationTypes from './rotation/types';
import * as SignalDetectionTypes from './signal-detection';
import * as ChannelTypes from './station-definitions/channel-definitions';
import * as ResponseTypes from './station-definitions/response-definitions';
import * as StationTypes from './station-definitions/station-definitions/station-definitions';
import * as ProcessingStationTypes from './station-processing/types';
import * as ReferenceStationTypes from './station-reference/types';
import * as TimeTypes from './time';
import * as TypeUtil from './type-util/type-util';
import * as ConfigurationTypes from './ui-configuration/types';
import * as UserProfileTypes from './user-profile/types';
import * as WaveformTypes from './waveform/types';
import * as WaveformUtil from './waveform/util';
import * as WorkflowTypes from './workflow/types';

export * from './faceted/utils';

export {
  ArrayUtil,
  BeamformingTemplateTypes,
  ChannelSegmentTypes,
  ChannelTypes,
  ColorTypes,
  CommonTypes,
  ConfigurationTypes,
  Displays,
  Endpoints,
  EventTypes,
  FacetedTypes,
  FilterTypes,
  FilterUtil,
  FkTypes,
  LegacyEventTypes,
  ProcessingMaskDefinitionTypes,
  ProcessingStationTypes,
  QcSegmentTypes,
  ReferenceStationTypes,
  ResponseTypes,
  RotationTypes,
  SignalDetectionTypes,
  StationTypes,
  TimeTypes,
  TypeUtil,
  UserProfileTypes,
  WaveformTypes,
  WaveformUtil,
  WorkflowTypes
};
