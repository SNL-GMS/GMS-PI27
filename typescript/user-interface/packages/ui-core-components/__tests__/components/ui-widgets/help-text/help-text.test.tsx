import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import { HelpText, HelpTextTarget } from '../../../../src/ts/components';

describe('HelpText', () => {
  it('matches a snapshot', () => {
    const { container } = render(
      <HelpTextTarget data-test-id="help-text-target">
        <HelpText>The contents of the help text tooltip...</HelpText>
        Hover over this to show icon
      </HelpTextTarget>
    );
    expect(container).toMatchSnapshot();
  });
  it('HelpTextTarget reveals the icon on hover', async () => {
    const { container } = render(
      <HelpTextTarget>
        <HelpText>The contents of the help text tooltip...</HelpText>
        <span data-test-id="hover-target">Hover target</span>
      </HelpTextTarget>
    );

    const hoverTarget = screen.getByText('Hover target');
    expect(hoverTarget).toBeInTheDocument();
    await userEvent.hover(hoverTarget);
    expect(screen.getByRole('img', { hidden: true })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
