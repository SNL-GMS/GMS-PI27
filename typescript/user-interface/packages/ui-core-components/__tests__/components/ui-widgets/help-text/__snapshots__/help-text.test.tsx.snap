// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText HelpTextTarget reveals the icon on hover 1`] = `
<div>
  <div
    class="help-text__hover-container"
  >
    <div
      aria-expanded="false"
      class="help-text bp5-popover-target"
    >
      <span
        aria-hidden="true"
        class="bp5-icon bp5-icon-info-sign"
        tabindex="0"
      >
        <svg
          data-icon="info-sign"
          height="14"
          role="img"
          viewBox="0 0 16 16"
          width="14"
        >
          <path
            d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zM7 3h2v2H7V3zm3 10H6v-1h1V7H6V6h3v6h1v1z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <span
      data-test-id="hover-target"
    >
      Hover target
    </span>
  </div>
</div>
`;

exports[`HelpText matches a snapshot 1`] = `
<div>
  <div
    class="help-text__hover-container"
  >
    <div
      aria-expanded="false"
      class="help-text bp5-popover-target"
    >
      <span
        aria-hidden="true"
        class="bp5-icon bp5-icon-info-sign"
        data-icon="info-sign"
        tabindex="0"
      />
    </div>
    Hover over this to show icon
  </div>
</div>
`;
